### PR TITLE
add Yukon (re #166)

### PR DESCRIPTION
**Pull request by jarek**
I found the data for Yukon's grid.

The grid covers only about 80% of Yukon's population, the off-grid population uses local diesel generators. I assumed that the off-grid population uses electricity at the same rate as the grid and added them in under oil. If you'd rather only include what is actually reported/what is on grid, I can remove the estimation of off-grid.

More info in the comment in CA_YT.py.

### CODE REVIEW

**Comment by corradio** [comment]
Related to #166

File: parsers/CA_YT.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    data = {
+        'datetime': datetime_datetime,
+        'countryCode': country_code,
+        'production': {
+            'oil': thermal_generation + off_grid_use,
+
+            'hydro': hydro_generation,
+
+            # specify some sources that aren't present in Yukon as zero,
+            # this allows the analyzer to better estimate CO2eq
```
Comment (corradio) [inline parsers/CA_YT.py]: 👍

File: parsers/CA_YT.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    if thermal_div.div:
+        thermal_text = hydro_div.div.text
+        thermal_generation = parse_mw(thermal_text)
+    else:
+        # thermal is not always used and when it's not used, it's not specified in HTML
+        thermal_generation = 0
+
+    # Yukon's grid covers about 80% of the population. The remainder has diesel generators.
+    # Assume that the off-grid uses electricity at same rate as grid. See longer comment above.
+    off_grid_use = (hydro_generation + thermal_generation) * 0.2
```
Comment (corradio) [inline parsers/CA_YT.py]: I'd rather have only the reported generation. Let's however keep the very detailed and good explanation you have. We might want to change it as some later stage.

File: parsers/CA_YT.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    of total population 35874 (2016 census), 28238 are in municipalities
+    that are connected to the grid - that is 78.7%.
+
+    We assume that the off-grid uses electricity at roughly the same rate as grid,
+    this assumption is probably a bit pessimistic but not too far off.
+    Off-grid generation is with diesel generators.
+
+    Yukon Energy reports only "hydro" vs "thermal" generation.
+    Per http://www.yukonenergy.ca/ask-janet/lng-and-boil-off-gas,
+    in 2016 the thermal generation was about 50% diesel and 50% LNG.
```
Comment (corradio) [inline parsers/CA_YT.py]: I normally count thermal as "unknown" (and the co2eq factors at 700g/kWh tend to represent that).
If it's quite certain that it's 50/50 diesel/gas then we can break it down, but I wouldn't assume it's all diesel (oil) if there's a substantial part that is gas.

Comment (jarek) [inline parsers/CA_YT.py]: My concern here is that using "unknown" (700) when it's known to be some mix of oil and gas (650 and 490) over-estimates the intensity a bit. But maybe it'd prompt the source to specify the breakdown more exactly...

**Comment by jarek** [comment]
@corradio changed the other things, let me know your preference on oil vs unknown

**Comment by corradio** [comment]
Just for reference, you said:
> My concern here is that using "unknown" (700) when it's known to be some mix of oil and gas (650 and 490) over-estimates the intensity a bit. But maybe it'd prompt the source to specify the breakdown more exactly...
I like to follow the principle that everything we state should be true, and that everything we're unsure about should be marked clearly as unknown. The fact that we do a mistake by estimating the value of unknown is ok in my opinion. However, what we can do is change, in `co2eq.js` the CO2 value of unknown, in order not to overestimate the gas/oil mix.
That seems like a good solution?

**Comment by jarek** [comment]
That makes sense, I made the change and I will include the Yukon diesel/LNG breakdown on my list of requests to various Canadian grid operators.

**Pull request merged**